### ISSUE TITLE
[Github CI] aarch64 (and possibly other archs) tests seem to be running in x64

### ISSUE THREAD

**Issue by eliphatfs** [vnmakarov/mir#251]
The defininitions for arch and distro is not used anywhere other than in name.

**Comment by vnmakarov** [comment]
Thank you for reporting this and all other issues.

You are right, my tests work on x86-64 only.  I'll fix it during the next week.  I expect there will be issues on different targets as they were not tested for long time.

**Comment by vnmakarov** [comment]
I tried to find does github actions provide real targets as it was before (or I thought so) but I failed to find this info.

I also tried to implement testing on github actions machines under qemu but it is too slow (testing arm64 takes more 2 hours).  I'll try to fix it in some other ways (I need time to explore self-hosting or other CI/CD providers) but it will definitely take more time.  Meanwhile, the testing will be done as it is now (all on x86-64).

I also found some test failures when I tried testing on real arm64.  I'll work on this too.

**Comment by vnmakarov** [comment]
I've switched aarch64 test to use self-hosting testing on an aarch64 machine.

There are still unresolved issue for testing for ppc64be/le and s390x.  Unfortunately, github does not provides software to do self-hosting testing on these architectures.  I'll continue to work on this issue.  I hope it will be resolved this spring.

**Comment by vnmakarov** [comment]
I've implemented ppc64 testing for github actions.  The testing is done on a ppc64 (power8) machine by using a self-hosting x86-64 machine.

I also implemented s390x testing.  It is done on a self-hosting x86-64 machine with qemu.  Therefore the testing is slow.  To speed up s390x testing, only tests for -O2 are run.

Btw, I am going to add riscv testing on qemu too.  But it will be not done soon.

So I believe this issue can be closed.
